src/backend/parser/parse_expr.c

```diff
@@ -931,7 +931,22 @@ rewrite_scope_identity_call(ParseState *pstate, Node **lexpr, Node **rexpr)
 	if (!(*lexpr) || !(*rexpr))
 		return;
 
-	if ((IsA(*lexpr, Var) && IsA(*rexpr, FuncExpr)))
+	if (IsA(*lexpr, Var) &&
+		IsA(*rexpr, FuncExpr) &&
+		strstr(get_func_name(((FuncExpr*) (*rexpr))->funcid), "like_escape") != NULL)
+    {
+        func_expr = (FuncExpr *) (*rexpr);
+        if((func_expr->args)->length==2){
+            Node *node = lsecond(func_expr->args);
+            if (IsA(node,Const) && ((Const *)(node))->constisnull)
```

Comment:
if it is an empty string, "", this value ((Const *)(node))->constisnull will still be false, I'm not sure if how do you check empty string.